components/empty-state/empty-state-illustrated-button.js

```diff
@@ -1,63 +1,139 @@
+ * @slot - Custom SVG content if `illustration-name` property is not set
  */
 class EmptyStateIllustratedButton extends RtlMixin(LitElement) {
 
 	static get properties() {
 		return {
 			/**
 			 * Optional: The action text to be used in the subtle button
-			 * @type {String}
+			 * @type {string}
 			 */
 			actionText: { type: String, attribute: 'action-text' },
 			/**
 			 * REQUIRED: A description giving details about the empty state
-			 * @type {String}
+			 * @type {string}
 			 */
 			description: { type: String },
 			/**
 			 * Optional: The name of the preset image you would like to display in the component
-			 * @type {String}
+			 * @type {string}
 			 */
-			illustration: { type: String },
+			illustrationName: { type: String, attribute: 'illustration-name' },
+			/**
+			 * Optional: This will change the action button to use a primary button instead of the default subtle button
+			 * @type {boolean}
+			 */
+			primary: { type: Boolean },
 			/**
 			 * REQUIRED: A title for the empty state
-			 * @type {String}
+			 * @type {string}
 			 */
-			title: { type: String }
+			titleText: { type: String, attribute: 'title-text' },
+			_contentWidth: { type: Number }
 		};
 	}
 
 	static get styles() {
-		return [emptyStateStyles, emptyStateIllustratedStyles, bodyCompactStyles, heading2Styles];
+		return [bodyCompactStyles, emptyStateStyles, emptyStateIllustratedStyles];
+	}
+
+	constructor() {
+		super();
+
+		this._contentWidth = 700;
+		this._resizeObserver = new ResizeObserver(this._onResize.bind(this));
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._resizeObserver.observe(this);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._resizeObserver.disconnect();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		if (!this.titleText || this.titleText.length === 0) {
+			console.warn('d2l-empty-state-illustrated-button component requires titleText.');
+		}
+
+		if (!this.description || this.description.length === 0) {
+			console.warn('d2l-empty-state-illustrated-button component requires a description.');
+		}
 	}
 
 	render() {
+		const illustrationContainerStyle = {
+			height: `${Math.min(this._contentWidth, 500) / 1.5}px`,
+		};
+
+		const titleSmall = this._contentWidth <= 615;
+		const titleClass = {
+			'd2l-empty-state-title': true,
+			'd2l-empty-state-title-small': titleSmall,
+			'd2l-empty-state-title-large': !titleSmall,
+		};
+
+		let actionButton = nothing;
+		if (this.actionText && this.primary) {
+			actionButton = this.actionText && html`<d2l-button class='d2l-empty-state-action' @click=${this._handleActionClick} primary>${this.actionText}</d2l-button>`;
+		}
+		else if (this.actionText && !this.primary) {
+			actionButton = this.actionText && html`<d2l-button-subtle class='d2l-empty-state-action' @click=${this._handleActionClick} text=${this.actionText}></d2l-button-subtle>`;
+		}
```

Comment:
Nit: could be simplified a bit I think:

```suggestion
		let actionButton = nothing;
		if (this.actionText) {
			actionButton = this.primary ? html`<d2l-button class='d2l-empty-state-action' @click=${this._handleActionClick} primary>${this.actionText}</d2l-button>` :
				html`<d2l-button-subtle class='d2l-empty-state-action' @click=${this._handleActionClick} text=${this.actionText}></d2l-button-subtle>`;
		}
```